### PR TITLE
Aborting Large Transactions

### PR DESCRIPTION
**Pull request by Maithem**
## Overview
When a transaction size exceeds the max limit, it used to fail with
a UnrecoverableCorfuError, this patch changes the behavior to throw
a TransactionAbortedException with AbortCause.SIZE_EXCEEDED.

Why should this be merged: Improves the transaction failure semantics

Related issue(s) (if applicable): #1609

## Checklist (Definition of Done):

- [x] There are no TODOs left in the code
- [x] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [x] Change is covered by automated tests
- [x] Public API has Javadoc

### CODE REVIEW

File: test/src/test/java/org/corfudb/integration/LargeWriteIT.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+/**
+ * A set integration tests that exercise failure modes related to
+ * large writes.
+ */
+
+public class LargeWriteIT extends AbstractIT {
+
+    @Test
+    public void largeStreamWrite() throws Exception {
+        final String host = "localhost";
```
Comment (xnull) [inline test/src/test/java/org/corfudb/integration/LargeWriteIT.java]: AbstractIT.DEFAULT_HOST

File: test/src/test/java/org/corfudb/integration/LargeWriteIT.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        final String streamName = "s1";
+        final int port = 9000;
```
Comment (xnull) [inline test/src/test/java/org/corfudb/integration/LargeWriteIT.java]: DEFAULT_HOST.DEFAULT_PORT

File: test/src/test/java/org/corfudb/integration/LargeWriteIT.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        final int maxWriteSize = 100;
+
+        // Configure a client with a max write limit
+        CorfuRuntime.CorfuRuntimeParameters params = CorfuRuntime.CorfuRuntimeParameters
+                .builder()
+                .maxWriteSize(maxWriteSize)
+                .build();
+
+        CorfuRuntime rt = CorfuRuntime.fromParameters(params);
+        rt.parseConfigurationString(host + ":" + port);
```
[the remark this answers is not in the document]
Comment (Maithem) [reply test/src/test/java/org/corfudb/integration/LargeWriteIT.java]: done.

File: test/src/test/java/org/corfudb/integration/LargeWriteIT.java [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        assertThatThrownBy(() -> rt.getStreamsView()
+                .get(CorfuRuntime.getStreamID(streamName))
+                .append(new byte[bufSize]))
+                .isInstanceOf(WriteSizeException.class);
+        shutdownCorfuServer(server_1);
+    }
+
+    @Test
+    public void largeTransaction() throws Exception {
+        final String host = "localhost";
```
[the remark this answers is not in the document]
Comment (Maithem) [reply test/src/test/java/org/corfudb/integration/LargeWriteIT.java]: done.

File: test/src/test/java/org/corfudb/integration/LargeWriteIT.java [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+        final int maxWriteSize = 100;
+
+        // Configure a client with a max write limit
+        CorfuRuntime.CorfuRuntimeParameters params = CorfuRuntime.CorfuRuntimeParameters
+                .builder()
+                .maxWriteSize(maxWriteSize)
+                .build();
+
+        CorfuRuntime rt = CorfuRuntime.fromParameters(params);
+        rt.parseConfigurationString(host + ":" + port);
```
[the remark this answers is not in the document]
Comment (Maithem) [reply test/src/test/java/org/corfudb/integration/LargeWriteIT.java]: done.

File: test/src/test/java/org/corfudb/integration/LargeWriteIT.java (hunk 2 above)
Comment (Maithem) [reply test/src/test/java/org/corfudb/integration/LargeWriteIT.java]: done.

File: test/src/test/java/org/corfudb/integration/LargeWriteIT.java (hunk 1 above)
Comment (Maithem) [reply test/src/test/java/org/corfudb/integration/LargeWriteIT.java]: done

**Pull request merged**